src/_guides/language/effective-dart/usage.md

```diff
@@ -352,6 +352,83 @@ var ints = new List<int>.from(numbers);
 But if your goal is just to copy the iterable and preserve its original type, or
 you don't care about the type, then use `toList()`.
 
+
+### DO use `whereType()` to filter a collection by type.
+
+Let's say you have a list containing a mixture of objects and you want to get just the integers out of it. You could do:
+
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/usage_bad.dart (where-type)"?>
+{% prettify dart %}
+var objects = [1, "a", 2, "b", 3];
+var ints = objects.where((e) => e is int);
+{% endprettify %}
+
+This is verbose, but, worse, it returns an Iterable whose type probably isn't
+what you want. In the example here, it returns an `Iterable<Object>` even though
+you likely want an `Iterable<int>` since that's the type you're filtering it to.
+
+Sometimes you see code that "corrects" the above error by doing:
+
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/usage_bad.dart (where-type-2)"?>
+{% prettify dart %}
+var objects = [1, "a", 2, "b", 3];
+var ints = objects.where((e) => e is int).cast<int>();
+{% endprettify %}
+
+That's verbose, and causes two wrappers to be created, with two layers of
+indirection and redundant runtime checking. Fortunately, the core library has a
+method for this exact use case:
+
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/usage_good.dart (where-type)"?>
+{% prettify dart %}
+var objects = [1, "a", 2, "b", 3];
+var ints = objects.whereType<int>();
+{% endprettify %}
+
```

Comment:
A few comments about this section:

1. I would add the three above excerpts to the tests file, and add a (hidden) test to ensure that the runtime type is `Iterable<Object>`, `Iterable<int>` and `Iterable<int>`, respectively.
2. Consider renaming `objects` to `iterable` (which was the name that you used in other examples involving iterables).
3. Consider linking to the API entry for `whereType()`:
> Fortunately, the core library has the [whereType()][] method for this exact use case:

    [whereType()]: {{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-core/Iterable/whereType.html

Reply:
> 1. I would add the three above excerpts to the tests file,

`whereType()` isn't actually fully supported yet (it doesn't work in Dart 1 mode). Is it OK if I don't worry about tests for this now. (For what it's worth, I find tests to be overkill for these tiny excerpts in Effective Dart. It takes much longer for me to make changes now and the tests and analysis have caught few issues for me.)

> 2. Consider renaming objects to iterable (which was the name that you used in other examples involving iterables).

I used the more specific names here because there are two iterables with different types in the excerpt.

> 3. Consider linking to the API entry for whereType():

Done.

Reply:
> Is it OK if I don't worry about tests for this now

Surely.